python/cudf/cudf/tests/test_doctests.py

```diff
@@ -0,0 +1,87 @@
+import doctest
+import inspect
+import os
+
+import numpy as np
+import pytest
+
+import cudf
+
+
+def _name_in_all(parent, name):
+    return name in getattr(parent, "__all__", [])
+
+
+def _is_public_name(parent, name):
+    return not name.startswith("_")
+
+
+def _find_doctests_in_obj(finder, obj, criteria):
+    """Find all doctests in an object.
+
+    Args:
+        finder (doctest.DocTestFinder): The DocTestFinder object to use.
+        obj (module or class): The object to search for docstring examples.
+        criteria (callable): Callable indicating whether to recurse over
+        members of the provided object.
+
+    Yields:
+        doctest.DocTest: The next doctest found in the object.
+    """
+    for docstring in finder.find(obj):
+        if docstring.examples:
+            yield docstring
+    for name, member in inspect.getmembers(obj):
+        # Only recurse over members matching the criteria
+        if not criteria(obj, name):
+            continue
+        # Recurse over the public API of modules (objects defined in the
+        # module's __all__)
+        if inspect.ismodule(member):
```

Comment:
I assume subpackages are also considered modules by this function?

Reply:
That is correct.